### ISSUE TITLE
update wcslib and cfitsio and leap second/IERS B table to the latest version

### ISSUE THREAD

**Issue by nden** [astropy/astropy#9018]
Edit: This is a perpetual reminder issue for doing updates for the actually upcoming release. Once addressed last before a release, remilestone it to the next one.

**Comment by pllim** [comment]
@nden, just for future clarity, what version do you mean? Do you plan to do it?

**Comment by nden** [comment]
A new one came out last week. I will include it in astropy soon but this issue concerns the future.

It is a reminder to check the latest wcslib that is out just before 4.0 is released. 
And yes, I will do it as usual.

**Comment by pllim** [comment]
Okay. v6.3 then -- https://www.atnf.csiro.au/people/mcalabre/WCS/wcslib/

**Comment by nden** [comment]
No, not really. I will update to 6.3 soon. This ticket is not about that. It is meant as a reminder to check what the latest version is before 4.0 is released.

**Comment by pllim** [comment]
Ah, okay. Thanks for the clarification!

**Comment by privong** [comment]
A note on the new wcslib, my distro (Arch Linux) has upgraded to version 6.3. I think that's caused `astropy.coordinates.SkyCoord` to no longer import successfully, throwing the following error:

```
  File "TNT-ALMA_LP.py", line 10, in <module>
    from astropy.coordinates import SkyCoord
  File "/usr/lib/python3.7/site-packages/astropy/coordinates/__init__.py", line 17, in <module>
    from .name_resolve import *
  File "/usr/lib/python3.7/site-packages/astropy/coordinates/name_resolve.py", line 21, in <module>
    from .sky_coordinate import SkyCoord
  File "/usr/lib/python3.7/site-packages/astropy/coordinates/sky_coordinate.py", line 11, in <module>
    from astropy.wcs.utils import skycoord_to_pixel, pixel_to_skycoord
  File "/usr/lib/python3.7/site-packages/astropy/wcs/__init__.py", line 26, in <module>
    from .wcs import *
  File "/usr/lib/python3.7/site-packages/astropy/wcs/wcs.py", line 49, in <module>
    from . import _wcs
ImportError: /usr/lib/libwcs.so.6: undefined symbol: fits_read_wcstab
```

If needed I can file a separate bug report.

**Comment by bsipocz** [comment]
I'm sorry for the quasi hijack, but cfitsio is in the same boat as wcslib, and this issue can actually serve as a reminder to check for updated versions for both prior the release (atm cfitsio is up-to-date, while wcslib already has a 6.4 version released a few weeks ago).

**Comment by bsipocz** [comment]
@nden - could you please open a PR for the wcslib update? I know we said "last before release", but would be nice to have the update in before RC1 (there is still some time until then, so no rush is needed).

**Comment by nden** [comment]
@bsipocz wcslib is up to date. If cfitsio does not need an update this ticket can be closed.

**Comment by bsipocz** [comment]
Added the leap second table to this list of things to update before a release.

cc @mhvk

**Comment by mhvk** [comment]
Let's be sure to also do `IERS A` - added that to the title here so that I can remove that on #10219 and there is only one "last before release" issue.

**Comment by eteq** [comment]
Note that there are no leap seconds to worry about in 2020 (the *only* thing not to worry about in 2020...), but I updated the IERS file for v4.1 in 9e8c3a48bc5cf12f127c5a6bca48edfaed4ca1fb.  with that I think this is all set.

**Comment by mhvk** [comment]
Indeed, leap seconds expire june 2021, so are ok.

**Comment by bsipocz** [comment]
Two things here: it's better to do PRs to the bugfix branch for patches like this, and keep this issue open and remilestone as it is relevant again for 4.3.

**Comment by nden** [comment]
#11260 updated the version of wcslib to v7.4. However, wcslib was patched to fix a[ minor problem](https://github.com/astropy/astropy/commit/d0a629ba36e1f8759d28f68d748835b9d04f65cf#diff-998c33c91f21460ae25bf3cf7396ff8d0bb66292cb220dfa8a5946811135037dR1949-R1950).

I don't know where else leave a record for the next update so I am adding a note here.

**Comment by dhomeier** [comment]
> #11260 updated the version of wcslib to v7.4. However, wcslib was patched to fix a[ minor problem](https://github.com/astropy/astropy/commit/d0a629ba36e1f8759d28f68d748835b9d04f65cf#diff-998c33c91f21460ae25bf3cf7396ff8d0bb66292cb220dfa8a5946811135037dR1949-R1950).
> 
> I don't know where else leave a record for the next update so I am adding a note here.

Note that the patch to WCSLIB itself was reverted in #11376 which instead fixed our build script to use the correct compiler flag. 
Upstream is now at v7.5; unfortunately there is still the excess warning from #11385.

**Comment by eteq** [comment]
done for 4.3 in 3aaad2220b0c0f85a7c526396b8c63bf154a1f87 , so I'm updating the milestone here.

**Comment by astrofrog** [comment]
@nden - should we have updated WCSLIB to 7.6 for the 5.0 release?

**Comment by dhomeier** [comment]
> @nden - should we have updated WCSLIB to 7.6 for the 5.0 release?

? 7.7 has already been merged in #12034 – may have forgotten to link this issue.

**Comment by astrofrog** [comment]
Ah great! Sorry I missed that PR somehow.

**Comment by pllim** [comment]
So, by now, we don't really need CFITSIO updates anymore.

@mhvk , do we still need leap second/IERS B updates now that we have `astropy-iers-data` package?

**Comment by saimn** [comment]
> So, by now, we don't really need CFITSIO updates anymore.

we can still get updates for fixes/changes in the few files we use:
https://github.com/astropy/astropy/tree/main/cextern/cfitsio/lib

**Comment by mhvk** [comment]
For IERS, I think this is indeed no longer needed!
